c7n/resources/apigw.py

```diff
@@ -793,3 +793,69 @@ def process(self, resources):
                     resourceId=m['resourceId'],
                     httpMethod=m['httpMethod'],
                     patchOperations=ops)
+
+
+@resources.register('apigw-domain-name')
+class CustomDomainName(query.QueryResourceManager):
+
+    class resource_type(query.TypeInfo):
+        enum_spec = ('get_domain_names', 'items', None)
+        arn = False
+        id = name = 'domainName'
+        service = 'apigateway'
+        universal_taggable = True
+        cfn_type = 'AWS::ApiGateway::DomainName'
+        date = 'createdDate'
+
+    @classmethod
+    def get_permissions(cls):
+        return ('apigateway:GET',)
+
+    @classmethod
+    def has_arn(self):
+        return False
+
+
+@CustomDomainName.filter_registry.register('check-tls')
+class DomainNameCheckTls(Filter):
+    """ Filter API Gateway custom domains and return those with security policy != TLS 1.2
+    """
+    permissions = ("apigateway:GET",)
+    schema = utils.type_schema('check-tls', )
+
+    def process(self, resources, event=None):
+        notenable_domains = []
+
+        for r in resources:
+            if r['securityPolicy'] != 'TLS_1_2':
+                notenable_domains.append(r)
+        return notenable_domains
+
+
+@CustomDomainName.action_registry.register('remediate-tls')
+class DomainNameRemediateTls(BaseAction):
```

Comment:
this feels like it should be renamed `update-security` .. and parameterized on tls version and cipher suites instead of hardcoded to make it of broader use.

Reply:
@kapilt I've updated the action based on your feedback